mz_crypt.c

```diff
@@ -34,11 +34,12 @@ int32_t mz_crypt_rand(uint8_t *buf, int32_t size) {
 
 uint32_t mz_crypt_crc32_update(uint32_t value, const uint8_t *buf, int32_t size) {
 #if defined(HAVE_ZLIB)
-   /* Define z_crc_t in zlib 1.2.5 and less or if using zlib-ng */
-#  if (ZLIB_VERNUM < 0x1270)
+#  if (ZLIB_VERNUM & 0xf != 0xf) && (ZLIB_VERNUM < 0x1270)
+    /* Define z_crc_t in zlib 1.2.6 and less */
     typedef unsigned long z_crc_t;
-#  else
-    typedef uint32_t z_crc_t;
+#  elif (ZLIB_VERNUM & 0xf == 0xf) && (ZLIB_VERNUM < 0x12df)
+    /* Define z_crc_t in zlib-ng 2.0.7 and less */
+    typedef unsigned int z_crc_t;
 #  endif
```

Comment:
z_crc_t was added in:
- zlib 1.2.7 (https://github.com/madler/zlib/commit/6c9bd474aa08312ef2e2e9655a80e18db24a1680) as `unsigned long`
- zlib-ng 2.1.0 (https://github.com/zlib-ng/zlib-ng/commit/ff9162c1a0da8d60ae664e5930b256de99ad83b3) as `unsigned int`

They were added with conflicting types, so we should not redefine them.